packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * Credit: Modified version of Plate's list plugin
+ * See: https://github.com/udecode/plate/blob/main/packages/nodes/list
+ */
+import {
+  getAbove,
+  getNode,
+  getPluginType,
+  insertNodes,
+  isLastChild,
+  PlateEditor,
+  TElement,
+} from '@udecode/plate-core';
+import { ELEMENT_LI, hasListChild, moveListItemsToList } from '@udecode/plate-list';
+import { Editor, NodeEntry, Path, Transforms } from 'slate';
+
+import { unwrapList } from './unwrapList';
+
+export interface MoveListItemUpOptions {
+  list: NodeEntry<TElement>;
+  listItem: NodeEntry<TElement>;
+}
+
+/**
+ * Move a list item up.
+ */
+export const moveListItemUp = (editor: PlateEditor, { list, listItem }: MoveListItemUpOptions) => {
+  const move = () => {
+    const [listNode, listPath] = list;
+    const [liNode, liPath] = listItem;
+
+    const liParent = getAbove<TElement>(editor, {
+      at: listPath,
+      match: { type: getPluginType(editor, ELEMENT_LI) },
+    });
+    if (!liParent) {
+      let toListPath;
+      try {
+        toListPath = Path.next(listPath);
+      } catch (err) {
+        return;
+      }
+
+      const condA = hasListChild(editor, liNode);
+      const condB = !isLastChild(list, liPath);
```

Comment:
What does it mean? Do we have better names?

Reply:
This file was copied as is from Plate. I made changes on the `moveListItemDown`

Reply:
I removed it completely and used the helper directly from Plate. I thought I'd need to modify it at first but not anymore.